### PR TITLE
779 opendap

### PR DESCRIPTION
**Pull request by gantian127**
Need front-end layout update before merge.

### CODE REVIEW

**Comment by hydrobuild** [comment]
Can one of the admins verify this patch?

File: hs_app_netCDF/page_processors.py [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-from crispy_forms.layout import Layout, Fieldset, ButtonHolder, Submit, HTML
-from forms import *
+
 from hs_core import page_processors
-from hs_core.forms import MetaDataElementDeleteForm
-from django.forms.models import formset_factory
 from hs_core.views import *
-from functools import partial, wraps
+
+from forms import *
```
Comment (ironfroggy) [inline hs_app_netCDF/page_processors.py]: We should avoid the use of `import *`

File: hs_app_netCDF/page_processors.py [hunk 2]
```diff
@@ -1,16 +1,15 @@
-__author__ = 'hydro'
+from crispy_forms.layout import Layout, Fieldset, ButtonHolder, Submit, HTML
```
Comment (ironfroggy) [inline hs_app_netCDF/page_processors.py]: This belongs after framework imports but before hydroshare imports.

File: hs_app_netCDF/page_processors.py (hunk 1 above)
Comment (ironfroggy) [inline hs_app_netCDF/page_processors.py]: Also avoid implicit relative imports. Either be explicit or use a full path.

File: hs_app_netCDF/templates/pages/netcdfresource.html [hunk 3]
```diff
@@ -180,17 +180,28 @@ <h3 style="margin-top:40px"> NetCDF Header Info</h3>
...
+{#    {% endif %}#}
+
+    {# Add THREDDS service #}
+    {% if nc_file_name %}
+        <div id="#thredds">
+            <h3>THREDDS Data Service</h3>
+            <h4>OPeNDAP: <a href={{ opendap }}> {{ nc_file_name }}</a></h4>
+            <h4>NetCDF Subset: <a href={{ ncss }}> {{ nc_file_name }}</a></h4>
+            <h4>NCML: <a href={{ ncml }}> {{ nc_file_name }}</a></h4>
+            <h4>ISO Metadata: <a href={{ iso }}> {{ nc_file_name }}</a></h4>
```
Comment (ironfroggy) [inline hs_app_netCDF/templates/pages/netcdfresource.html]: these `href` attributes should be quoted.

File: hs_app_netCDF/templates/pages/netcdfresource.html [hunk 4]
```diff
@@ -180,17 +180,28 @@ <h3 style="margin-top:40px"> NetCDF Header Info</h3>
...
-        <td><a style="margin-left:15px" id="btn-netcdf-opendap" type="button" class="btn btn-success" href='{{opendap_link }}'
-               title="When changing this resource's sharing status, please allow 2 minutes for the OPeNDAP service to refresh on this page" >
-            <span class="button-label"> OPeNDAP Service</span>
-        </a></td>
+{#    {% if thredds_service %}#}
+{#        <td><a style="margin-left:15px" id="btn-netcdf-opendap" type="button" target='_blank'class="btn btn-success" href='{{thredds_service}}'#}
+{#               title="When changing this resource's sharing status, please allow 2 minutes for the THREDDS data service to refresh on this page" >#}
+{#            <span class="button-label"> THREDDS Data Services </span>#}
+{#        </a></td>#}
+{#    {% endif %}#}
```
Comment (ironfroggy) [inline hs_app_netCDF/templates/pages/netcdfresource.html]: If this is not needed, it should be removed.

File: hs_app_netCDF/templates/pages/netcdfresource.html [hunk 5]
```diff
@@ -226,12 +237,12 @@ <h3 style="margin-top:40px"> NetCDF Header Info</h3>
...
-        if ($('#btn-netcdf-opendap').length > 0){
-           $("#btn-netcdf-opendap").insertAfter("#btn-download-all");
-        }
-    })
+{#    // netcdf opendap button#}
+{#    $(document).ready(function(){#}
+{#        if ($('#btn-netcdf-opendap').length > 0){#}
+{#           $("#btn-netcdf-opendap").insertAfter("#btn-download-all");#}
+{#        }#}
+{#    })#}
```
Comment (ironfroggy) [inline hs_app_netCDF/templates/pages/netcdfresource.html]: Remove this if it isn't needed.

File: hs_app_netCDF/page_processors.py [hunk 6]
```diff
@@ -115,13 +114,21 @@ def landing_page(request, page):
...
         for f in content_model.files.all():
             if '.nc' in f.resource_file.name[-3:]:
-                context['opendap_link'] = 'http://hydrotest41.renci.org:8080/thredds/dodsC/HFCat/{0}/data/contents/{1}.html'.\
-                    format(content_model.short_id, f.resource_file.name.split('/')[-1])
+                ip = 'http://thredds.hydroshare.org'
+                shortkey = content_model.short_id
+                nc_file_name = f.resource_file.name.split('/')[-1]
+
+                context['opendap'] = '{}/thredds/dodsC/HFCat/{}/data/contents/{}.html'.format(ip, shortkey, nc_file_name)
+                context['ncss'] = '{}/thredds/ncss/HFCat/{}/data/contents/{}/dataset.html'.format(ip,shortkey, nc_file_name)
```
Comment (hyi) [inline hs_app_netCDF/page_processors.py]: add a space before "shortkey"

**Comment by gantian127** [comment]
@ironfroggy @hyi I just updated the code, please have a check and let me know if the code needs edits. Thanks.

File: hs_app_netCDF/templates/pages/netcdfresource.html [hunk 7]
```diff
@@ -179,18 +179,21 @@ <h3 style="margin-top:40px"> NetCDF Header Info</h3>
...
-    {% if opendap_link %}
-        <td><a style="margin-left:15px" id="btn-netcdf-opendap" type="button" class="btn btn-success" href='{{opendap_link }}'
-               title="When changing this resource's sharing status, please allow 2 minutes for the OPeNDAP service to refresh on this page" >
-            <span class="button-label"> OPeNDAP Service</span>
-        </a></td>
+    {# Add THREDDS service #}
+    {% if nc_file_name %}
+        <div id="#thredds">
+            <h3>THREDDS Data Service</h3>
+            <h4>OPeNDAP: <a href="{{ opendap }}" target="_blank"> {{ nc_file_name }}</a></h4>
```
Comment (hyi) [inline hs_app_netCDF/templates/pages/netcdfresource.html]: it'd be good to add a space between {{ nd_file_name }} and </a> for consistency. Same applies to the following few statements as well, very minor formatting comment as a FYI.

**Comment by ironfroggy** [comment]
+1

File: hs_app_netCDF/templates/pages/netcdfresource.html (hunk 7 above)
Comment (gantian127) [inline hs_app_netCDF/templates/pages/netcdfresource.html]: @hyi I will update the {{nc_file_name}} format. Mauriel will help with the front-end layout update after Thur. I think after he finishes the update, this branch can be merged back. Thanks.

**Comment by gantian127** [comment]
@hyi Would you please merge this branch so that this can be tested on production site? I think I will create a new issue for the front-end layout update which will be implemented by Mauriel after AGU. Thank you.

**Comment by hyi** [comment]
@gantian127 Merged it to develop. Will update beta so that you can test it out before dev is fixed to be back up.

**Pull request merged**